[AIRFLOW-3761] Decommission User & Chart models & Update doc accordingly

XD-DENG opened the pull request:
### Jira

  - https://issues.apache.org/jira/browse/AIRFLOW-3761

### Description

#### Background
In master branch, we have already decommissioned the Flask-Admin UI.

#### What have been done
- In model definitions, User and Chart are only applicable for the "old" UI based on Flask-Admin. Hence we should decommission these two models as well.

- Alembic updated accordingly

- Related doc are updated in this commit as well (like `profiling` section is no longer applicable).

ashb commented on airflow/config_templates/default_airflow.cfg:
```diff
@@ -271,9 +271,6 @@ access_logfile = -
 # Expose the configuration file in the web server
-# This is only applicable for the flask-admin based web UI (non FAB-based).
-# In the FAB-based web UI with RBAC feature,
-# access to configuration is controlled by role permissions.
```
If this is only applicable to the old UI why do we keep the setting?

XD-DENG replied:
In 1.10.2, I remember I found that config was hidden even if for Admin. So I suspect that `expose_config` is also applicable for RBAC UI now (possibly due to some new commit). If you check the codebase, "expose_config" is still used somewhere.

So I didn't remove this.

XD-DENG replied:
Double-checked: This `expose_config` item is used by the current UI (RBAC) now. We should keep it and remove the out-dated comment as I did.

XD-DENG committed:
[AIRFLOW-3761] Decommission User & Chart models & Update doc accordingly

In master branch, we have already decommissioned the Flask-Admin UI.

In model definitions, User and Chart are only applicable for the
"old" UI based on Flask-Admin.
Hence we should decommission these two models as well.

Related doc are updated in this commit as well.

ashb reviewed (approved):
LGTM. @bolkedebruin any other comments?
